nixos/modules/services/misc/mautrix-whatsapp.nix

```diff
@@ -0,0 +1,100 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mautrix-whatsapp;
+
+  configFile = pkgs.runCommand "mautrix-whatsapp" {
+    buildInputs = [ pkgs.mautrix-whatsapp pkgs.remarshal ];
+    preferLocalBuild = true;
+  } ''
+    mkdir -p $out
+    remarshal -if json -of yaml \
+      < ${pkgs.writeText "config.json" (builtins.toJSON cfg.configOptions)} \
+      > $out/config.yaml
+
+    ${pkgs.mautrix-whatsapp}/bin/mautrix-whatsapp -c $out/config.yaml -g -r $out/registration.yaml
+  '';
+
+in
+{
+  options.services.mautrix-whatsapp = {
+    enable = mkEnableOption {
+      type = types.bool;
```

Comment:
ofborg provides the following error:

```
cannot coerce a set to a string
```

This happens because `mkEnableOption` expects a string what to enable:

``` nix
mkEnableOption "mautrix-whatsapp"
```

The generated option is already false by default and has the type `bool`. If you want to alter the description you can do something like this:

``` nix
mkEnableOption "mautrix-whatsapp" // {
  description = ''
    Description.
  '';
}
```

Reply:
Thanks, I've some other module that I did which are not on this pattern ..